src/common/snippets/src/lowered/tensor.cpp

```diff
@@ -0,0 +1,105 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "snippets/lowered/tensor.hpp"
+
+#include <snippets/itt.hpp>
+#include "snippets/utils.hpp"
+
+
+namespace ngraph {
+namespace snippets {
+namespace lowered {
+
+Tensor::Tensor(const ExpressionPort& source_descriptor, const std::vector<ExpressionPort>& consumer_descriptors)
+    : m_source_port(source_descriptor), m_consumer_ports(consumer_descriptors) {}
+
+std::vector<ExpressionPort>::const_iterator Tensor::find_consumer(const ExpressionPort& consumer) const {
+    // Note: Find by shared ptr and index port is enough since these parameters must be unique
+    return std::find_if(m_consumer_ports.begin(), m_consumer_ports.end(),
+                        [&consumer](const ExpressionPort& td) {
+                            return consumer.get_expr_ptr() == td.get_expr_ptr() && consumer.get_index() == td.get_index();
+                        });
+}
+
+std::vector<ExpressionPort>::iterator Tensor::find_consumer(const ExpressionPort& consumer) {
+    // Note: Find by shared ptr and index port is enough since these parameters must be unique
+    return std::find_if(m_consumer_ports.begin(), m_consumer_ports.end(),
+                        [&consumer](const ExpressionPort& td) {
+                            return consumer.get_expr_ptr() == td.get_expr_ptr() && consumer.get_index() == td.get_index();
+                        });
+}
+
+bool Tensor::found_consumer(const ExpressionPort& consumer) const {
+    return find_consumer(consumer) != m_consumer_ports.end();
+}
+
+void Tensor::add_consumer(const ExpressionPort& consumer) {
+    OPENVINO_ASSERT(!found_consumer(consumer), "Consumer has been already added to Tensor!");
+    m_consumer_ports.push_back(consumer);
+}
+
+void Tensor::remove_consumer(const ExpressionPort& consumer) {
+    const auto& found = find_consumer(consumer);
+    OPENVINO_ASSERT(found != m_consumer_ports.end(), "Consumer is missed in Tensor!");
+    m_consumer_ports.erase(found);
+}
+
+std::vector<ExpressionPort> Tensor::get_conflicted_consumers() const {
+    std::vector<ExpressionPort> conflicted_consumers;
+    for (const auto& consumer : m_consumer_ports) {
+        if (is_conflicted_consumer(consumer)) {
+            conflicted_consumers.push_back(consumer);
+        }
+    }
+    return conflicted_consumers;
+}
+
+bool Tensor::is_conflicted_consumer(const ExpressionPort& consumer) const {
```

Comment:
1) Minor: `is_conflicting_consumer`, also true for get...
2) The conflict is determined by the loop insertion logic, and it has nothing to do with `Tensor` itself. `Tensor` - is just an entity representing a memory blob. Moreover, subtensor and layout should not be the `Tensor` properties, they should belong to respective ports. So the logic that interprets the port contents (compares the ports connected to this `Tensor`) should be inside  the mark_loops pass, not here.